include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp

```diff
@@ -123,11 +182,55 @@ class affine_cell_proxy : public tuple_t
     //!\overload
     decltype(auto) vertical_score() const && noexcept
     { //Unfortunately gcc7 does not preserve the const && type: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=94967
-        using return_t = std::tuple_element_t<2, tuple_t>;
+        using return_t = std::tuple_element_t<2, score_cell_type>;
         return static_cast<return_t const &&>(get_score_impl<2>(std::move(*this)));
     }
     //!\}
 
+    /*!\name Trace value accessor
+     * \brief Specific accessor function to get the respective trace value from an affine matrix cell.
+     * \{
+     */
+    //!\brief Access the optimal score of the wrapped score matrix cell.
+    decltype(auto) best_trace() & { return get_trace_impl<0>(*this); }
+    //!\overload
+    decltype(auto) best_trace() const & { return get_trace_impl<0>(*this); }
+    //!\overload
+    decltype(auto) best_trace() && { return get_trace_impl<0>(std::move(*this)); }
+    //!\overload
+    decltype(auto) best_trace() const &&
+    { //Unfortunately gcc7 does not preserve the const && type: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=94967
+        using return_t = std::tuple_element_t<0, trace_cell_type>;
+        return static_cast<return_t const &&>(get_trace_impl<0>(std::move(*this)));
+    }
+
+    //!\brief Access the horizontal score of the wrapped score matrix cell.
+    decltype(auto) horizontal_trace() & { return get_trace_impl<1>(*this); }
+    //!\overload
+    decltype(auto) horizontal_trace() const & { return get_trace_impl<1>(*this); }
+    //!\overload
+    decltype(auto) horizontal_trace() && { return get_trace_impl<1>(std::move(*this)); }
+    //!\overload
+     decltype(auto) horizontal_trace() const &&
```

Comment:
There seems to be an unnecessary space here :)
```suggestion
    decltype(auto) horizontal_trace() const &&
```